From c687a0a4b765d58bf8b68f1120492844a6f6d294 Mon Sep 17 00:00:00 2001
From: yingbinx <yingbinx.zeng@intel.com>
Date: Wed, 13 Sep 2017 16:40:07 +0800
Subject: [PATCH 07/17] storage: update rpmb sim write

update rpmb sim write to fix failure of
save data after reboot.

Jira: None
Test: Device boots to home screen.
Change-Id: I856073f48c20723b96aa6a38724cb5e95d3c0373
Signed-off-by: Huang, Yang <yang.huang@intel.com>
Signed-off-by: yingbinx <yingbinx.zeng@intel.com>
---
 trusty/storage/proxy/storage.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/trusty/storage/proxy/storage.c b/trusty/storage/proxy/storage.c
index 7e6c4a4..89c7317 100644
--- a/trusty/storage/proxy/storage.c
+++ b/trusty/storage/proxy/storage.c
@@ -136,6 +136,11 @@ ssize_t write_with_retry(int fd, const void *buf_, size_t size, off_t offset)
         buf += rc;
         offset += rc;
     }
+    rc = fsync(fd);
+    if (rc < 0) {
+        ALOGE("fsync for fd=%d failed: %s\n", fd, strerror(errno));
+        return rc;
+    }
     return 0;
 }
 
-- 
2.7.4

